lib/Helper/CircleHelper.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Tables\Helper;
+
+use OCA\Circles\CirclesManager;
+use OCA\Circles\Model\Member;
+use OCA\Circles\Model\Probes\CircleProbe;
+use OCP\App\IAppManager;
+use OCP\Server;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+/**
+ * @psalm-suppress UndefinedClass
+ */
+class CircleHelper {
+	private LoggerInterface $logger;
+	private bool $circlesEnabled;
+	private ?CirclesManager $circlesManager;
+
+	/**
+	 * @psalm-suppress UndefinedClass
+	 */
+	public function __construct(
+		LoggerInterface $logger,
+		IAppManager     $appManager
+	) {
+		$this->logger = $logger;
+		$this->circlesEnabled = $appManager->isEnabledForUser('circles');
+		$this->circlesManager = null;
+
+		if ($this->circlesEnabled) {
+			try {
+				$this->circlesManager = Server::get(CirclesManager::class);
+			} catch (Throwable $e) {
+				$this->logger->warning('Failed to get CirclesManager: ' . $e->getMessage());
+				$this->circlesEnabled = false;
+			}
+		}
+	}
```

Comment:
```suggestion
	private bool $circlesEnabled;
	private ?CirclesManager $circlesManager;

	/**
	 * @psalm-suppress UndefinedClass
	 */
	public function __construct(
		private LoggerInterface $logger,
		IAppManager $appManager
	) {
		$this->circlesEnabled = $appManager->isEnabledForUser('circles');
		$this->circlesManager = null;

		if ($this->circlesEnabled) {
			try {
				$this->circlesManager = Server::get(CirclesManager::class);
			} catch (Throwable $e) {
				$this->logger->warning('Failed to get CirclesManager: ' . $e->getMessage());
				$this->circlesEnabled = false;
			}
		}
	}
```

nitpick: use property promotion